tests/test_gsd.py

```
@@ -149,5 +149,5 @@ def test_gsd_update_and_report():
 
 def test_gsd_simulate_invalid_sims():
     design = GroupSequentialDesign(k=3)
-    with pytest.raises(ValueError, match="Number of simulations must be positive"):
+    with pytest.raises(ValueError, match="Number of simulations must be a positive integer."):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)